### PR TITLE
BoS Crafting Changes

### PR DESCRIPTION
**Pull request by BobNewman-max**
## About The Pull Request
Major crafting changes, if I fucked it up. Let me know.

Shifts crafting in the BoS to the scribes to give them more reason to exist. Knights no longer get round-start books 3/4 knowledge. 

SKs can no longer craft M1 Carbines, I've been told they're good (Even though I know they're not) so they get hunting shotguns instead. 

HS gets a box which has 2 crafting recipes which are decent. They're a decent (ish) listing. They also have a chance to get an AER9/AEP7 blueprint which are there to be throw-away BPs. If they get them they're honestly gonna be useless but that's the point. They have a chance to get shit that is shit.

Fixed the codex that's in-game. Not going to add the link in there as it starts now round-start on their screen. When anyone in the BoS joins it gives them the link to the servers codex in their chat. Because no one else could be bothered to do it.

The codex in-game change is done due to the players for some reason refusing to check discord at times and also (same people) screaming at me for the in-game one being bad. On the discord. Funny.

All-in-all its done to shift crafting towards scribes rather then knights.

Paladin also gets their power-fist because Imma be real, 2 melee is stupid. Especially since the hammer is much better for damage. So why not just let them keep the good stuff. (I'll add something in the future if it seems like a pointless loadout.)

Fixed a comment that I put incorrectly about the SP getting an AER14. They're supposed to have an AER12. 

Back to the crafting changes, 

Scribes and Senior Scribes get GnB PT.1-4 learned. Because they craft stuff now.
HS gets a box of rando BPs (2) and SS can choose it as a loadout. SS also lose their 6 research blueprints and now have 3. Might reduce it further we'll see.

Knights also  get an engineer loadout. So. Fun. 

## Pre-Merge Checklist
- [x] You tested this on a local server.
- [x] This code did not runtime during testing.
- [x] You documented all of your changes.

### CODE REVIEW

**Comment by ErwinSnowden** [comment]
This doesn't work as intended and the "Knight-Engineer" loadout just serves to invalidate the change you made to "take G&B part 3/4 from knights", making this a direct buff.

image
Ah yes, roundstart RCW/riot shotgun blueprints... something you can only get in two of the hardest bunkers. The LSW and R84 blueprints which are both LMGs, VERY balanced. And... the infiltrator... which is a unique? Infinite uniques. Cool.

Not to mention that you also gave the senior knight the ability to craft the hunting shotgun over the M1 carbine, yet another inarguable buff since the M1 carbine shoots ten millimeter.

Beyond the fact that this doesn't work because you don't understand the GLOBAL_INIT_LIST, **hard no**. The Brotherhood already had a massive buff.

**Comment by BadAtThisGame302** [comment]
If your previous PR of adding another paladin wasn't powercreep... then this PR definitely IS powercreep.

**Comment by BobNewman-max** [comment]
the box that gives them half decent weapons to craft with only 2 good guns is a power-creep, okay. Homie only the RCW is cursed, that and the R84 are powerful. But yeah I can remove the two OP ones B)

**Comment by ErwinSnowden** [comment]
The infiltrator apparently uses "SEMI_AUTO_NODELAY", the only obtainable gun outside of a toy/the bow (singleshot) which does so.

It's basically on par with the sequoia and a unique handed only to the NCR lieutenant, you _can_ get it from the unique lootdrops though. I see no issue with giving it to potentially every Brotherhood member.

**Comment by BobNewman-max** [comment]
> 

didn't notice, yeah I'll change that B)

File: code/modules/jobs/job_types/bos.dm [hunk 1]
```diff
@@ -600,10 +599,16 @@ Senior Scribe
...
+	H.mind.teach_crafting_recipe(/datum/crafting_recipe/wattz1k)
+	H.mind.teach_crafting_recipe(/datum/crafting_recipe/magnetowattz)
 	H.mind.teach_crafting_recipe(/datum/crafting_recipe/booster)
 	ADD_TRAIT(H, TRAIT_CHEMWHIZ, src)
 	ADD_TRAIT(H, TRAIT_SURGERY_HIGH, src)
 	ADD_TRAIT(H, TRAIT_CYBERNETICIST, src)
+	ADD_TRAIT(H, TRAIT_GUNSMITH_ONE, src)
+	ADD_TRAIT(H, TRAIT_GUNSMITH_TWO, src)
+	ADD_TRAIT(H, TRAIT_GUNSMITH_THREE, src)
+	ADD_TRAIT(H, TRAIT_GUNSMITH_FOUR, src)
```
Comment (OliOliOnsiPree) [inline code/modules/jobs/job_types/bos.dm:611]: for(var/datum/crafting_recipe/recipe as() in GLOB.knight_gunsmith_recipes)
		H.mind.teach_crafting_recipe(recipe)

(Thats how they get the recipes via one to four)

File: code/modules/jobs/job_types/bos.dm [hunk 2]
```diff
@@ -712,10 +717,15 @@ Scribe
...
 	H.mind.teach_crafting_recipe(/datum/crafting_recipe/AER9)
+	H.mind.teach_crafting_recipe(/datum/crafting_recipe/AEP7)
 	H.mind.teach_crafting_recipe(/datum/crafting_recipe/booster)
 	ADD_TRAIT(H, TRAIT_SURGERY_MID, src)
 	ADD_TRAIT(H, TRAIT_MEDICALGRADUATE, src)
 	ADD_TRAIT(H, TRAIT_CHEMWHIZ, src)
+	ADD_TRAIT(H, TRAIT_GUNSMITH_ONE, src)
+	ADD_TRAIT(H, TRAIT_GUNSMITH_TWO, src)
+	ADD_TRAIT(H, TRAIT_GUNSMITH_THREE, src)
+	ADD_TRAIT(H, TRAIT_GUNSMITH_FOUR, src)
```
Comment (OliOliOnsiPree) [inline code/modules/jobs/job_types/bos.dm:728]: for(var/datum/crafting_recipe/recipe as() in GLOB.knight_gunsmith_recipes)
		H.mind.teach_crafting_recipe(recipe)

File: code/modules/jobs/job_types/bos.dm [hunk 3]
```diff
@@ -279,6 +279,7 @@ Head Scribe
 		/obj/item/clothing/mask/gas/adv/gasmask = 1,
 		/obj/item/stock_parts/cell/ammo/ec = 2,
 		/obj/item/gun/energy/laser/plasma/glock = 1,
+		/obj/item/storage/box/bos = 1
```
Comment (OliOliOnsiPree) [inline code/modules/jobs/job_types/bos.dm:282]: if you plan on giving the box to head scribe you might aswell give them the recipes that 1 - 4 get, but thats majority up to you. 

	for(var/datum/crafting_recipe/recipe as() in GLOB.knight_gunsmith_recipes)
		H.mind.teach_crafting_recipe(recipe)

**Review by OliOliOnsiPree — CHANGES_REQUESTED** [review]
There you go.

**Comment by ErwinSnowden** [comment]
This still doesn't work because knights don't lose their 3/4 knowledge, because they don't have separate recipe lists. 💀

**Comment by Legate-Maxson** [comment]
Do to community reaction this will be closed for now.

**Pull request closed without merging**